samples/matter/common/src/app/fabric_table_delegate.h

```diff
@@ -57,10 +61,18 @@ class AppFabricTableDelegate : public chip::FabricTable::Delegate {
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 				chip::DeviceLayer::ThreadStackMgr().ClearAllSrpHostAndServices();
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
-				/* Erase Matter data */
-				chip::DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().DoFactoryReset();
 				/* Erase Network credentials and disconnect */
 				chip::DeviceLayer::ConnectivityMgr().ErasePersistentInfo();
+#ifdef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL
+				/* Disable mpsl before start removing settings to improve the performance. */
+				mpsl_lib_uninit();
```

Comment:
As disucssed F2F, let's remove it from here and just modify Matter Platform's code.

Reply:
Provided requested changes.